src/app/pages/questions/services/healthkit.service.ts

```diff
@@ -65,25 +69,18 @@ export class HealthkitService {
   loadData(healthDataType) {
     return this.getLastPollTimes().then(dic => {
       let lastPollTime = this.MIN_POLL_TIMESTAMP
-      if (healthDataType in dic) lastPollTime = dic[healthDataType]
+      // if (healthDataType in dic) lastPollTime = dic[healthDataType]
       return CapacitorHealthkit
         .requestAuthorization(
           {
             all: [''],
-            read: [healthDataType], //read only permission
+            read: this.READ_PERMISSIONS,
```

Comment:
Permission screen will only be shown once. If the user denies HK permissions, there will be no data in the query results. Maybe we can explicitly tell the user to make sure this is enabled in the Health app.

Reply:
Yes, when we do not get any data from query, we can say something like. `We did not receive any data from Apple Health. Can you please check that permission in the Apple Health app are enabled for RADAR Questionnaire app. <Add detailed instructions>`

Reply:
Okay sounds good. I will add the changes in a separate PR.